packages/clarity-decode/src/clarity.ts

```diff
@@ -18,8 +18,10 @@ export function decode(input: string): DecodedPayload {
     let envelope = data.envelope(json.e);
     let timestamp = Date.now();
     let payload: DecodedPayload = { timestamp, envelope };
+
     // Sort encoded events by time to simplify summary computation
-    let encoded: Data.Token[][] = json.d.sort((a: Data.Token[], b: Data.Token[]): number => (a[0] as number) - (b[0] as number));
+    let encoded: Data.Token[][] = json.p ? json.a.concat(json.p) : json.a;
+    encoded = encoded.sort((a: Data.Token[], b: Data.Token[]): number => (a[0] as number) - (b[0] as number));
```

Comment:
Minor: Why do you need to sort them? Maybe add comment.

Reply:
Because different providers (e.g. pointer vs. resize) can buffer events at a different rate and even though Clarity ensures timestamps are valid - i.e. when the event happened - they may not be able to go on wire at the exact time. We ensure that we relatively sort all the events before processing them further or visualizing them.

Reply:
Please add the above in comment. 

---
In reply to: [467635660](https://github.com/microsoft/clarity/pull/31#discussion_r467635660) [](ancestors = 467635660)